src/components/UserInfoForm/UserInfoForm.tsx

```diff
@@ -0,0 +1,336 @@
+    formState: { errors, isValid },
+  } = useForm<IFormInput>({ mode: "all" });
+
+  const dispatch = useDispatch();
+
+  const FormSubmitHandler: SubmitHandler<IFormInput> = (data: IFormInput) => {
+    history.push("./loading");
+    dispatch(addUserInfoToReduxAction(data));
+  };
+
+  return (
+    <div className={formClasses.bgMain}>
+      <div className={formClasses.root}>
+        <div>
+          <Header />
+          <form onSubmit={handleSubmit(FormSubmitHandler)}>
+            <Box
+              component="form"
+              sx={{
+                "& .MuiTextField-root": { m: 0.8, width: "17ch" },
+              }}
+              noValidate
+              autoComplete="off"
+            >
+              <TextField
+                error={!!errors.firstName}
+                id="outlined-size-normal"
+                label="First Name"
+                className={formClasses.Fname}
+                {...register("firstName", { required: true })}
+                onInput={(evt: any) => {
+                  const caps =
+                    evt.target.value.charAt(0).toUpperCase() +
+                    evt.target.value.slice(1);
+                  setValue("firstName", caps);
+                }}
+                helperText={
+                  !!errors.firstName && <p>First Name is Required!</p>
+                }
+              />
+              <TextField
+                error={!!errors.lastName}
+                id="outlined-size-normal"
+                label="Last Name"
+                className={formClasses.Lname}
+                {...register("lastName", { required: true })}
+                onInput={(evt: any) => {
+                  const caps =
+                    evt.target.value.charAt(0).toUpperCase() +
+                    evt.target.value.slice(1);
+                  setValue("lastName", caps);
+                }}
+                helperText={!!errors.lastName && <p>Last Name Is Required!</p>}
+              />
+            </Box>
+            <TextField
+              error={!!errors.PhoneNumber}
+              fullWidth
+              type="number"
+              id="outlined-basic"
+              label="Phone Number"
+              variant="outlined"
+              className={formClasses.TxtBottom}
+              {...register("PhoneNumber", { required: true })}
+              helperText={
+                !!errors.PhoneNumber && <p>Phone Number Is Required!</p>
+              }
+              InputProps={{
+                startAdornment: (
+                  <InputAdornment position="start">
+                    <Button
+                      onClick={() => setOpen(true)}
+                      className={formClasses.CountryBtnPos}
+                    >
+                      <img
+                        src={country.icon}
+                        alt={country.value}
+                        className={formClasses.CountryImg}
+                      />
+                    </Button>
+                    <Dialog
```

Comment:
why did you place it under adornment prop?

Reply:
Dialog box is placed at the bottom of the last div of this component